app/Listeners/Network/RecordFirEntry.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace App\Listeners\Network;
+
+use App\Events\NetworkAircraftUpdatedEvent;
+use App\Models\FlightInformationRegion\FlightInformationRegion;
+use App\Models\Vatsim\NetworkAircraftFirEvent;
+use App\Services\FlightInformationRegionService;
+use Location\Coordinate;
+use Location\Polygon;
+
+class RecordFirEntry
+{
+    /**
+     * @var Polygon[]
+     */
+    private $boundaries;
+
+    public function __construct()
+    {
+        $this->boundaries = FlightInformationRegion::all()
+            ->mapWithKeys(function (FlightInformationRegion $region) {
+                return [$region->id => FlightInformationRegionService::getBoundaryPolygon($region->identification_code)];
+            })
+            ->toArray();
+    }
+
+    public function handle(NetworkAircraftUpdatedEvent $event): bool
+    {
+        $lastEntry = NetworkAircraftFirEvent::where('callsign', $event->getAircraft()->callsign)
+            ->where('event_type', 'FIR_ENTRY')
+            ->latest('created_at')
+            ->first();
+
+        // Handle FIR exit
+        if (
+            $lastEntry &&
+            !$this->hasPreviouslyExitedFir($event->getAircraft()->callsign, $lastEntry->flight_information_region_id) &&
+            isset($this->boundaries[$lastEntry->flight_information_region_id]) &&
```

Comment:
Can you define this as variable to make it a bit more readable? Maybe something like `$isWithinBoundary`

Reply:
Done!